### ISSUE TITLE
[🐛] 🔥 auth().useEmulator doesn't work with IP hosts

### ISSUE THREAD

**Issue by olivierlesnicki** [invertase/react-native-firebase#4878]
## Issue

I've launched an auth emulator on port 9099.

Everything works fine on the iOS simulator when my config is as follow:
`auth().useEmulator('http://localhost:9099')`

Auth services stop working on the iOS simulator when I provide my Mac IP instead:
`auth().useEmulator('http://192.168.0.160:9099')`
and return an `[auth/network-request-failed]` error

Auth services don't work at all on my device with either configuration, 
resulting in the `[auth/network-request-failed]` error

### Javascript

<details><summary>Click To Expand</summary>
<p>

#### `package.json`:

```json
# N/A
```

#### `firebase.json` for react-native-firebase v6:

```json
# N/A
```

</details>

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [ ] my application is an AndroidX application?
- [ ] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

</p>
</details>

---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
System:
    OS: macOS 10.15.7
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 19.99 MB / 32.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.15.1 - /usr/local/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.14.8 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.10.0 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.4, DriverKit 20.2, macOS 11.1, tvOS 14.3, watchOS 7.2
    Android SDK:
      API Levels: 28, 29
      Build Tools: 28.0.3, 29.0.2, 29.0.3
      System Images: android-22 | Google APIs ARM EABI v7a, android-22 | Google APIs Intel x86 Atom_64, android-24 | Google Play Intel x86 Atom, android-28 | Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom, android-29 | Google APIs Intel x86 Atom, android-R | Google Play Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 3.6 AI-192.7142.36.36.6392135
    Xcode: 12.4/12D4e - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_232 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.13.1 => 16.13.1 
    react-native: 0.63.4 => 0.63.4 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [ ] Android
  - [x] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `10.5.1`
- **`Firebase` module(s) you're using that has the issue:**
  - `@react-native-firebase/auth`
- **Are you using `TypeScript`?**
  - `N`

</p>
</details>

---

- 👉 Check out [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter for updates on the library.

**Comment by olivierlesnicki** [comment]
After investigation I found an issue upstream in Firebase iOS SDK:
https://github.com/firebase/firebase-ios-sdk/issues/7342

**Comment by mikehardy** [comment]
Looks like it will be fixed in firebase-ios-sdk 7.6.0, nothing to do but wait I suppose? No action possible here, I will resolve this as blocked upstream - @ziyafenn - look for the 7.6.0 ios sdk release from firebase and re-test your case as well? Report back here if not fixed and we may reopen

**Comment by mikehardy** [comment]
@olivierlesnicki / @ziyafenn firebase-ios-sdk 7.6.0 was just released, I'll do a react-native-firebase 10.7.0 shortly that will have it.

**Comment by olivierlesnicki** [comment]
@mikehardy thanks
